bot/cogs/moderation.py

```diff
@@ -875,121 +884,68 @@ def mod_log(self) -> ModLog:
-                "If `expires_at` is given as a non-datetime, "
-                "it must be `permanent`."
-            )
-        if expires_at == 'permanent':
-            expires_at = None
-
-        try:
-            previous_infraction = await self.bot.api_client.get(
-                'bot/infractions/' + str(infraction_id)
-            )
-
-            # check the current active infraction
-            infraction = await self.bot.api_client.patch(
-                'bot/infractions/' + str(infraction_id),
-                json={
-                    'expires_at': (
-                        expires_at.isoformat()
-                        if expires_at is not None
-                        else None
-                    )
-                }
-            )
-
-            # Re-schedule
-            self.cancel_task(infraction['id'])
-            loop = asyncio.get_event_loop()
-            self.schedule_task(loop, infraction['id'], infraction)
-
-            if expires_at is None:
-                await ctx.send(f":ok_hand: Updated infraction: marked as permanent.")
-            else:
-                human_expiry = (
-                    datetime
-                    .fromisoformat(infraction['expires_at'][:-1])
-                    .strftime('%c')
-                )
-                await ctx.send(
-                    ":ok_hand: Updated infraction: set to expire on "
-                    f"{human_expiry}."
-                )
-
-        except Exception:
-            log.exception("There was an error updating an infraction.")
-            await ctx.send(":x: There was an error updating the infraction.")
-            return
-
-        # Get information about the infraction's user
-        user_id = infraction["user"]
-        user = ctx.guild.get_member(user_id)
-
-        if user:
-            member_text = f"{user.mention} (`{user.id}`)"
-            thumbnail = user.avatar_url_as(static_format="png")
-        else:
-            member_text = f"`{user_id}`"
-            thumbnail = None
-
-        # The infraction's actor
-        actor_id = infraction["actor"]
-        actor = ctx.guild.get_member(actor_id) or f"`{actor_id}`"
+        if expires_at is None and reason is None:
+            # Unlike UserInputError, the error handler will show a specified message for BadArgument
+            raise BadArgument("Neither a new expiry nor a new reason was specified.")
+
+        # Retrieve the previous infraction for its information.
+        old_infraction = await self.bot.api_client.get(f'bot/infractions/{infraction_id}')
+
+        request_data = {}
+        confirm_messages = []
+        log_text = ""
+
+        if expires_at == "permanent":
+            request_data['expires_at'] = None
+            confirm_messages.append("marked as permanent")
+        elif expires_at is not None:
+            request_data['expires_at'] = expires_at.isoformat()
+            confirm_messages.append(f"set to expire on {expires_at.strftime('%c')}")
```

Comment:
In a slightly, let's call it "interesting", vote and debate with the moderation team, we decided that we want to display infraction datetime strings in an ISO-like string (`"%Y-%m-%d %H:%M"`), like here:

https://github.com/python-discord/bot/blob/1685d676008d3092fa1e2377769613a5063e10f8/bot/cogs/moderation.py#L1132

This has not been consistently changed throughout the cog, yet, and many occurrences of `%c` still exist. Nevertheless, I think we should strive for consistency and use the newly chosen format here as well.

```suggestion
            confirm_messages.append(f"set to expire on {expires_at.strftime('"%Y-%m-%d %H:%M"')}")
```

Reply:
I could just knock these all out in this PR cause it's only for moderation stuff, right?

Reply:
Done, for the most part. All infraction time formats should now be consistent. The only exception is for watch channels, which has a similar format but with seconds included too. Was not sure if that should be touched.

Reply:
It can be replaced, as far as I'm concerned. There's a utility function in the `WatchChannel` ABC that's used by the talentpool cog for a display that's very similar to the infraction search output. Originally, I had the idea of promoting that local utility function into a general one, but you've already taken care of that now.

The only thing left is testing, but you've probably thought of that as well. I guess we need to wait for #421 for that.

Reply:
I've replaced it by the new utility function.